### PR TITLE
feat: add the BE requests to receive messages, implement Asset and Event Api

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

#115 #51 #137 #142

----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: src/main/java/com/wire/kalium/cli/CliSendApplication.kt
Comment (gongracr) [inline src/main/java/com/wire/kalium/cli/CliSendApplication.kt:143]: Weird indentation again? 🤔

**Review by gongracr — APPROVED** [review]
Looks good enough for now to start with the project structure reshaping. Great work Mo 🍪

File: src/main/java/com/wire/kalium/cli/CliSendApplication.kt
Comment (vitorhugods) [reply src/main/java/com/wire/kalium/cli/CliSendApplication.kt:143]: Hmm 🤔 weird indeed.

Did you run Command + L or is it leftover from before the restyle?

Comment (MohamadJaara) [reply src/main/java/com/wire/kalium/cli/CliSendApplication.kt]: fixed, i did run the Command + l and it fixed it 
i think it's from before the restyle

Comment (MohamadJaara) [reply src/main/java/com/wire/kalium/cli/CliSendApplication.kt]: i will do a rebase before merging just to make sure it's all good

**Comment by gongracr** [comment]
You can squash and merge this PR @MohamadJaara, unless you guys have a different accept/merge flow?

**Pull request merged**
